app/models/environment.rb

```diff
@@ -25,6 +26,15 @@ def active_lock
     locks.active.first
   end
 
+  # Configures this environment to auto deploy the given branch.
+  def configure_auto_deploy(branch, options = {})
+    self.update_attributes!(auto_deploy_branch: branch, auto_deploy_user: options[:user])
```

Comment:
Awesome, thanks for the info @atmos. Yeah, I think we can easily do properly attributed git authors here, with a fallback to a bot user if the author has never logged into /deploy.

2fa is something that I hadn't put much thought into, but it's a great point. I'll add an issue for that.